src/test/java/io/supertokens/test/bulkimport/apis/AddBulkImportUsersTest.java

```diff
@@ -0,0 +1,408 @@
+            }
+        }
+        // No two loginMethods can have isPrimary as true
+        {
+            // CASE 1: email, passwordHash and hashingAlgorithm are not present
+            try {
+                JsonObject request = new JsonParser()
+                        .parse("{\"users\":[{\"loginMethods\":[{\"recipeId\":\"emailpassword\",\"email\":\"johndoe@gmail.com\",\"passwordHash\":\"somehash\",\"hashingAlgorithm\":\"bcrypt\",\"isPrimary\":true},{\"recipeId\":\"passwordless\",\"email\":\"johndoe@gmail.com\",\"isPrimary\":true}]}]}").getAsJsonObject();
+                HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                        "http://localhost:3567/bulk-import/add-users",
+                        request, 1000, 1000, null, Utils.getCdiVersionStringLatestForTests(), null);
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                String responseString = getResponseMessageFromError(e.getMessage());
+                assertEquals(400, e.statusCode);
+                assertEquals(responseString,
+                        "{\"error\":\"" + genericErrMsg +  "\",\"users\":[{\"index\":0,\"errors\":[\"No two loginMethods can have isPrimary as true.\"]}]}");
+            }
+        }
+        // Can't import than 10000 users at a time
+        {
+            try {
+                JsonObject request = generateUsersJson(10001);
+            HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+            "http://localhost:3567/bulk-import/add-users",
+            request, 1000, 1000, null, Utils.getCdiVersionStringLatestForTests(), null);
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                String responseString = getResponseMessageFromError(e.getMessage());
+                assertEquals(400, e.statusCode);
+                assertEquals(responseString, "{\"error\":\"You can only add 1000 users at a time.\"}");
+            }
+        }
+
+        process.kill();
+        Assert.assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void shouldReturn200Response() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        JsonObject request = generateUsersJson(10000);
+        JsonObject response = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+        "http://localhost:3567/bulk-import/add-users",
+        request, 1000, 1000, null, Utils.getCdiVersionStringLatestForTests(), null);
+        assertEquals("OK", response.get("status").getAsString());
```

Comment:
We should have also checked the records in DB. How do we do this usually? Should I create a TestOnly method in Storage to get the users? 

In future, I will add an API to get bulk import users, that can be used as well.